discovery-infra/tests/base_test.py

```diff
@@ -15,70 +15,70 @@
 from junit_report import JunitFixtureTestCase, JunitTestCase
 from paramiko import SSHException
 from test_infra import consts
-from tests.config import ClusterConfig
+from tests.config import ClusterConfig, EnvConfig
 from test_infra.controllers.proxy_controller.proxy_controller import ProxyController
 from test_infra.helper_classes.cluster import Cluster
 from test_infra.helper_classes.kube_helpers import create_kube_api_client, KubeAPIContext
 from test_infra.helper_classes.nodes import Nodes
 from test_infra.tools.assets import NetworkAssets
+from tests.config import TerraformConfig
 from tests.conftest import env_variables
+from test_infra.controllers.node_controllers import TerraformController
 
 
 class BaseTest:
-    @staticmethod
-    def override_node_parameters(**kwargs):
-        _vars = deepcopy(env_variables)
-        for key, value in kwargs.items():
-            _vars[key] = value
 
-        _vars['num_nodes'] = _vars['num_workers'] + _vars['num_masters']
+    @pytest.fixture(scope="function")
+    def get_nodes(self) -> Callable:
+        """ Currently support only single instance of nodes """
+        nodes_data = dict()
 
-        return _vars
+        @JunitTestCase()
+        def get_nodes_func(config: TerraformConfig = TerraformConfig()):
+            if "nodes" in nodes_data:
+                return nodes_data["nodes"]
 
```

Comment:
What does it mean?

Reply:
That for leaving the current logic as is.
Pytest cached the yield value from fixtures. On `nodes` fixture it cached the actual `Nodes` object and on `get_nodes` fixture the returned callback is cached.